contrib/mesos/pkg/scheduler/components/scheduler.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"net/http"
+	"sync"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"k8s.io/kubernetes/contrib/mesos/pkg/backoff"
+	"k8s.io/kubernetes/contrib/mesos/pkg/offers"
+	"k8s.io/kubernetes/contrib/mesos/pkg/queue"
+	"k8s.io/kubernetes/contrib/mesos/pkg/runtime"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/algorithm"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/algorithm/podschedulers"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/binder"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/deleter"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/errorhandler"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/framework"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/podreconciler"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/schedulerloop"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/config"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/podtask"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/queuer"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/record"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+// sched implements the Scheduler interface.
+type sched struct {
+	podReconciler podreconciler.PodReconciler
+	framework     framework.Framework
```

Comment:
I would argue that it's not really another layer. It's just a different
interface. And a much smaller one in scope than Framework. It would make me
feel better about the giant interface that framework.Framework exposes

On Wed, Nov 4, 2015 at 10:53 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/components/scheduler.go
> <https://github.com/kubernetes/kubernetes/pull/16316#discussion_r43897272>
> :
>
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/framework"
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/podreconciler"
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/components/schedulerloop"
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/config"
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/podtask"
> > +	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/queuer"
> > +	"k8s.io/kubernetes/pkg/api"
> > +	"k8s.io/kubernetes/pkg/client/cache"
> > +	"k8s.io/kubernetes/pkg/client/record"
> > +	client "k8s.io/kubernetes/pkg/client/unversioned"
> > +)
> > +
> > +// sched implements the Scheduler interface.
> > +type sched struct {
> > +	podReconciler podreconciler.PodReconciler
> > +	framework     framework.Framework
>
> The sched struct implements an interface already for that reason, i.e. to
> be able to mock it. I would avoid another layer of interfaces until we need
> that structural complexity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16316/files#r43897272>.
>